core/unit_test/UnitTest_ScopeGuard.cpp

```diff
@@ -0,0 +1,123 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <cstdlib>
+#include <gtest/gtest.h>
+#include <Kokkos_Core.hpp>
+
+/**
+ * Fixture that checks Kokkos is neither initialized nor finalized before and
+ * after the test.
+ */
+class AssertEnvironmentTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    ASSERT_FALSE(Kokkos::is_initialized());
+    ASSERT_FALSE(Kokkos::is_finalized());
+  }
+
+  void TearDown() override {
+    ASSERT_FALSE(Kokkos::is_initialized());
+    ASSERT_FALSE(Kokkos::is_finalized());
+  }
+};
+
+namespace Test {
```

Comment:
Please make it the anonymous namespace 
```suggestion
namespace {
```
and move the definition of `AssertEnvironmentTest` into it